src/goto-programs/remove_complex.cpp

```diff
@@ -105,20 +104,22 @@ static void remove_complex(exprt &expr)
 
   if(expr.id()==ID_typecast)
   {
-    assert(expr.operands().size()==1);
-    if(expr.op0().type().id()==ID_complex)
+    auto const &typecast_expr = to_typecast_expr(expr);
+    if(typecast_expr.op().type().id() == ID_complex)
     {
-      if(expr.type().id()==ID_complex)
+      if(typecast_expr.type().id() == ID_complex)
       {
         // complex to complex
       }
       else
       {
         // cast complex to non-complex is (T)__real__ x
         unary_exprt tmp(
-          ID_complex_real, expr.op0(), expr.op0().type().subtype());
+          ID_complex_real,
+          typecast_expr.op(),
+          typecast_expr.op().type().subtype());
```

Comment:
We now have a `complex_real_exprt`, please use this one in here while changing those lines.